pkg/root/signing_config.go

```diff
@@ -46,48 +47,89 @@ type ServiceConfiguration struct {
 	Count    uint32
 }
 
+// uint32Slice implements sort.Interface methods for a uint32 slice
+type uint32Slice []uint32
+
+func (s uint32Slice) Len() int           { return len(s) }
+func (s uint32Slice) Less(i, j int) bool { return s[i] < s[j] }
+func (s uint32Slice) Swap(i, j int)      { s[i], s[j] = s[j], s[i] }
+
 // SelectService returns which service endpoint should be used based on supported API versions
-// and current time. It will select the first service that matches the criteria. Services should
-// be sorted from newest to oldest validity period start time, to minimize how far clients
-// need to search to find a matching service.
+// and current time. It will select the first service with the highest API version that matches
+// the criteria. Services should be sorted from newest to oldest validity period start time, to
+// minimize how far clients need to search to find a matching service.
 func SelectService(services []Service, supportedAPIVersions []uint32, currentTime time.Time) (string, error) {
-	for _, s := range services {
-		if slices.Contains(supportedAPIVersions, s.MajorAPIVersion) && s.ValidAtTime(currentTime) {
-			return s.URL, nil
+	if len(supportedAPIVersions) == 0 {
+		return "", fmt.Errorf("no supported API versions")
+	}
+
+	sortedVersions := make([]uint32, len(supportedAPIVersions))
+	copy(sortedVersions, supportedAPIVersions)
+	sort.Sort(sort.Reverse(uint32Slice(sortedVersions)))
```

Comment:
You can use the generic slices package to remove `uint32Slice`:

```go
	slices.Sort(sortedVersions)
	slices.Reverse(sortedVersions)
```

Reply:
Thanks, updated!

Reply:
Seems not to be updated, maybe you didn't push?

Reply:
I think a git reset and git push -f wiped out a commit, oops! Added back.